dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageResource.java

```diff
@@ -479,7 +475,7 @@ private Response getPageRender(final PageRenderParams renderParams) throws DotDa
      */
     private void setUpTimeMachineIfPresent(final PageRenderParams renderParams) {
         final Optional<Instant> timeMachineDate = renderParams.timeMachineDate();
-        if(timeMachineDate.isPresent()){
+        if(timeMachineDate.isPresent() && isOlderThanGraceWindow(timeMachineDate.get())){
```

Comment:
The validation should happen when we build the render params object.
If the date is invalid, it doesn't get included in the RenderParams.

If we do it at the session initialization moment we are saying the renderParams Object has the time-machine Param but somehow it never made it to the session set. it's confusing.  I'd rather have clearer logic like This: we never set the Date in the PageRenderParams object unless it is valid. Also if the date escapes a valid score we need to log that info